src/rest/src/main/java/org/geoserver/rest/RestConfiguration.java

```diff
@@ -133,7 +134,9 @@ protected void addInterceptors(InterceptorRegistry registry) {
         registry.addInterceptor(new RestInterceptor());
         registry.addInterceptor(new CallbackInterceptor());
     }
-
+    // ContentNegotiationConfigurer.favorPathExtension is deprecated because Spring wants to
```

Comment:
In cases where it can't match on the extension I think it returns a default, see https://www.baeldung.com/spring-mvc-content-negotiation-json-xml  I don't think it falls back to accepts.